bokeh/models/tools.py

```diff
@@ -371,8 +371,22 @@ class HoverTool(Tool):
     off by setting ``tooltips=None``.
 
     .. warning::
-        Point hit testing is not currently available on all glyphs. Hover tool
-        currently does not work with line or image type glyphs.
+        Hover tool does not currently work with the following glyphs:
+            * annulus
+            * arc
+            * bezier
+            * gear
+            * image
+            * image_rgba
+            * image_url
+            * multi_line
+            * oval
+            * patch
+            * quad
```

Comment:
I don't see that error. The problem is we don't actually check/enforce that `top > bottom` and `right > left`. If I make this change:
```
diff --git a/examples/glyphs/glyphs.py b/examples/glyphs/glyphs.py
index 571645f..6c24b5f 100644
--- a/examples/glyphs/glyphs.py
+++ b/examples/glyphs/glyphs.py
@@ -52,7 +52,7 @@ glyphs = [
     ("oval", Oval(x="x", y="y", width=screen(15), height=screen(25), angle=-0.7, fill_color="#1D91C0")),
     ("patch", Patch(x="x", y="y", fill_color="#A6CEE3")),
     ("patches", Patches(xs="xs", ys="ys", fill_color="#FB9A99")),
-    ("quad", Quad(left="x", right="xm01", top="y", bottom="ym01", fill_color="#B3DE69")),
+    ("quad", Quad(left="x", right="xp01", top="y", bottom="ym01", fill_color="#B3DE69")),
     ("quadratic", Quadratic(x0="x", y0="y", x1="xp02", y1="y", cx="xp01", cy="yp01", line_color="#4DAF4A", line_width=3)),
     ("ray", Ray(x="x", y="y", length=45, angle=-0.7, line_color="#FB8072", line_width=2)),
     ("rect", Rect(x="x", y="y", width=screen(10), height=screen(20), angle=-0.7, fill_color="#CAB2D6")),
``` 
the hover tool works. I would suggest making that change in this PR. 

For consistency, probably ought to do this somewhere soon too:
```
diff --git a/bokehjs/src/coffee/renderer/glyph/quad.coffee b/bokehjs/src/coffee/renderer/glyph/quad.coffee
index 6eb0f64..fcced82 100644
--- a/bokehjs/src/coffee/renderer/glyph/quad.coffee
+++ b/bokehjs/src/coffee/renderer/glyph/quad.coffee
@@ -36,8 +36,8 @@ class QuadView extends Glyph.View
 
     hits = []
     for i in [0...@sleft.length]
-      if (sx >= @sleft[i] and sx <= @sright[i] and sy >= @stop[i] and
-          sy < @sbottom[i])
+      console.log @sleft[i], sx, @sright[i], @stop[i], sy, @sbottom[i]
+      if (sx >= @sleft[i] and sx <= @sright[i] and sy >= @stop[i] and sy <= @sbottom[i])
```